src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -1375,11 +1379,27 @@ public String getParamDocName(String var) {
     return localVarName(Name.from(var));
   }
 
+  /** Returns the language specific format of parameter documentation. */
+  public String getParamDocText(String paramName, String paramTypeName, String text) {
+    return getNotImplementedString("SurfaceNamer.getParamDocText");
+  }
+
   /** Converts the given text to doc lines in the format of the current language. */
   public List<String> getDocLines(String text) {
     return CommonRenderingUtil.getDocLines(commentReformatter.reformat(text));
   }
 
+  /**
+   * Converts the given text to doc lines and split at maxWidth. If reformat is true, text is also
+   * reformatted in the format of the current language.
```

Comment:
Looking at https://github.com/googleapis/gapic-generator/blob/master/src/main/java/com/google/api/codegen/util/java/JavaCommentReformatter.java, I see reformatting  also includes, eg, escaping  special characters.

Are we guaranteed this reformatting only happens once, total, for each language? (eg all the `getParamDocText` already reformat....) If we reformat twice, the Java escape codes will be incorrect: `<` → `&lt;`→`&amp;lt;`, which would render as `&lt;` rather than `<` in JavaDoc. (tl;dr: reformatting is not idempotent)

Naïvely, it seems to me we would always format or never format..... It seems like in your usages, it's always false, so do we ever need it to be true? If not, I suggest removing the parameter, and then we don't have to worry about the idempotency issue.

Reply:
I mainly added `boolean reformat` because the other overload (a few lines above) always reformats the text, and if we don't pass in a flag here, it'd be confusing that these two methods have different behaviors.

Reply:
I think then we're mixing semantics. All the other `getDocLines` unconditionally format and don't worry about `maxWidth`. Why does this method need to be an overload? It seems to me it's not used in the same situations as the others (is that right? you always call it with `reformat==false`). How about  calling `wrapDocLines(String doc, int maxWidth)` and saying `// Wraps the given formatted doc text at maxWidth columns`?

Reply:
Ah I see.

I _think_ this is commonly used style in Java. Since overloading methods in Java is fairly easy and Java is verbose enough, it's very common to reuse method names (to describe the major task) with different parameters (to describe details), as the parameters are most of the time self-documenting. So it's common to have a method with only the most crucial parameters take care of the default / most common behavior, and have overloads take care of customized behaviors.

But it's just a style problem, not something that needs to be strictly enforced. In fact I think changing to `wrapDocLines` makes sense too and I'll just do that.